src/boot/CompressedAsyncStorage.js

```diff
@@ -0,0 +1,46 @@
+/* @flow */
+import { AsyncStorage, NativeModules } from 'react-native';
+
+export default class CompressedAsyncStorage {
+  // HACK: This wrapper around AsyncStorage.getItem() is not perfectly compliant with the
+  // specification of AsyncStorage.getItem(): `callback` should be optional, and getItem()
+  // should return the result. Here, we declare `callback` as non-optional, because redux-persist
+  // always calls getItem() with a callback. redux-persist is the only library that uses
+  // CompressedAsyncStorage. Also, we don't return the result, because redux-persist never uses it
+  // (instead, it looks at the result passed to `callback`). This avoids decompressing the result
+  // twice.
```

Comment:
What if we say something like
```
const outerResult = await NativeModules.TextCompressionModule.decompress(result);
callback(error, outerResult);
return outerResult;
```
?

I think that'd let us return the result without extra work.

If we really can't, then that's fine, though.

Reply:
Ah, right, that'd work!